### PR TITLE
Add doxygen doc in nav2_lifecycle_manager header files

### PR DESCRIPTION
**Pull request by shivaang12**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |

---

## Description of contribution in a few bullet points
* Add doc to nav2_lifecycle_manager

---

## Future work that may be required in bullet points
* Need review!
* Its kinda hard for me to put into words whats happening in both the classes, so need suggestions for the class brief

### CODE REVIEW

**Comment by SteveMacenski** [comment]
I'm not terribly familiar with this package, I'll have to ask someone else to give you the reviews

**Comment by shivaang12** [comment]
@mjeronimo thanks for the review!

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp [hunk 1]
```diff
@@ -31,11 +31,20 @@ namespace nav2_lifecycle_manager
 {
 
 using nav2_msgs::srv::ManageLifecycleNodes;
-
+/**
+ * @class nav2_lifecycle_manager::LifecycleManager
+ * @brief Implements lifecycle node manager service
```
[the remark this answers is not in the document]
Comment (shivaang12) [reply nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp]: How about this?
```
/**
 * @class nav2_lifecycle_manager::LifecycleManager
 * @brief Implements service interface to control the lifecycle nodes of
 * Navigation2 stack. It receives control request and then uses lifecycle
 * interface to change lifecycle node's state.
 */
```

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp [hunk 2]
```diff
@@ -29,30 +29,80 @@
...
-
+/**
+ * @enum nav2_lifecycle_manager::SystemStatus
+ * @brief Enum class representing the status of the system.
+ */
 enum class SystemStatus {ACTIVE, INACTIVE, TIMEOUT};
-
+/**
+ * @class nav2_lifecycle_manager::LifeCycleMangerClient
+ * @brief Implements lifecycle manager service client and NavigateToPose action
```
[the remark this answers is not in the document]
Comment (shivaang12) [reply nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp]: Is this okay to say?
```
/**
 * @class nav2_lifecycle_manager::LifeCycleMangerClient
 * @brief Implements lifecycle manager service client and NavgationToPose
 * action client.
 */
```

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp [hunk 3]
```diff
@@ -29,30 +29,80 @@
...
 {
 public:
+  /**
+   * @brief A constructor for LifeCycleMangerClient
+   */
   LifecycleManagerClient();
 
   // Client-side interface to the Nav2 lifecycle manager
+  /**
+   * @brief Makes start up service call
```
Comment (bpwilcox) [inline nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp]: I would align same grammar in this file as you used in the `lifecycle_manager.hpp` (e.g. "**Make** start up service call").

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp [hunk 4]
```diff
@@ -31,11 +31,22 @@ namespace nav2_lifecycle_manager
 {
 
 using nav2_msgs::srv::ManageLifecycleNodes;
-
+/**
+ * @class nav2_lifecycle_manager::LifecycleManager
+ * @brief Implements service interface to control the lifecycle nodes of
+ * Navigation2 stack. It receives control request and then uses lifecycle
```
Comment (bpwilcox) [inline nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp]: I think the word **transition** is more appropriate in this case as opposed to control (e.g. "...to **transition** the lifecycle nodes" and "It receives **transition** request...").

**Comment by shivaang12** [comment]
@bpwilcox Thanks for the review!

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (mjeronimo) [reply nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp]: How about: "The LifecycleManagerClient sends requests to the LifecycleManager to control the lifecycle state of the navigation modules."

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjeronimo) [reply nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp]: Yes, sounds good.

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (shivaang12) [reply nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager_client.hpp]: Sounds plausible!

**Comment by crdelsey** [comment]
@shivaang12 Is this ready to go in?

**Comment by shivaang12** [comment]
@crdelsey yes! Please go ahead and merge it.

**Pull request merged**